### ISSUE TITLE
"Unable to read some of your folders" / "Conv. Rekeying required"

### ISSUE THREAD

**Issue by Marenz** [keybase/keybase-issues#3112]
Moved from https://github.com/keybase/keybase-issues/issues/3074

I connected 2 new computers and validated them with my phone. One of my conversations now says I need to rekey it. Upon pressing I see 
![image](https://user-images.githubusercontent.com/424752/32751829-5bc3713c-c8c7-11e7-943b-0866295e5544.png)
The `Moto` device is however turned on and the chat is available on that.
I saw the same message box on the other computer. Upon entering a paper key they disappeared.. at first... actually let me bring some order in the events that happened:

* Install keybase on device "Toliop"
* Install keybase on device "edox"
* Verify "Toliop" using "Moto"
* Verify "edox" using "Moto"
* See pictured msg box on both new devices
* Create Paper Key on "moto"
* Enter Paper Key on "Toliop" and "edox"
* Box disappears on both
* Click on private conversation on "Toliop"
* See "REKEY NEEDED" next to the user name. Conversation itself not visible, only the red msg "This conversation needs to be rekeyed"
* Click on "Rekey"
* See the same message box as before again.
* Go here and search & post about the issue.

log ID for `edox` is `d745a5462b00ca3a1efbb71c`
log ID for `Toliop` is `bc66cae2732e1b861ba01a1c`

**Comment by strib** [comment]
Thanks!  It might also be useful if you could send feedback from `Moto`, including. Use the "..." menu to find the option.

**Comment by Marenz** [comment]
Done. I was not provided a log ID but I referenced this issues' id in the description

**Comment by strib** [comment]
Thanks, the ID was a9d0539ee98a33e589e2471c.  Looking.

**Comment by strib** [comment]
Hmm, it seems like `Moto` doesn't have a live connection to one of our servers, though unfortunately the logs from the phone don't go far back enough to tell why that might be.  It hasn't contacted this particular server in about 2 hours, since before you added those new devices I think.

As a quick fix, restarting Keybase on `Moto`, or launching Keybase on `SociomanticPC` should key the folder for you.  If it doesn't, please send feedback again from `Moto`.

**Comment by Marenz** [comment]
Restarting the App on `Moto` did indeed help. Thank you for the quick reaction. Too bad we couldn't find the source of the bug yet.

**Comment by strib** [comment]
So weird.  I guess we need to extend the amount of logging we buffer on mobile.  Sorry for the bad experience, glad it's working now.

**Comment by Marenz** [comment]
What would be the symptoms I'd see if it happens again? Does it only become apparent when I link new devices?

**Comment by strib** [comment]
Yeah, it would only happen when you add new devices, and the other participants in the chat/folder aren't online to rekey the folder for you (as happened in your other chat).

**Comment by Marenz** [comment]
Okay, that sounds difficult to recreate. I probably will not have that case soon again.

**Comment by Marenz** [comment]
I seem to keep having the issue that the App disconnects from the server and fails to reconnect. Right now a friend wrote something in a team chat. I got the notification on my mobile phone but I didn't see any message. Trying to write something stays "Grey" (I guess that means unconfirmed?) however, checking my desktop client shows that my message was actually sent.

Earlier I had something similar, but then it said "failed to send message 'retry'". Restarting the App helped.

I also don't see my messages that I typed from the desktop.
I've sent another feedback & logs mentioning this issue.

**Comment by DeepMac** [comment]
Have the same issue on my Windows PC install of Keybase, have had it since I did a reset. Used  my paper key for the reset, which seemed to work correctly.
Only one chat conversation has this issue for me. On my Android install I can still communicate with the other user.

Here's the relevant line from my logs where it throws:
018-02-24T17:43:36.947728-05:00 - [DEBU keybase tlf.go:165] e1f3 ++Chat: - KBFSNameInfoSource: CryptKeys(tlf=[removed],mode=CHAT_GUI) -> ERROR: This device does not yet have read access to directory /keybase/private/[removed], log into Keybase from one of your other devices to grant access: could not find params for uid=1b8a8e11cad14f4a503bbea20c918319 device 

followed by a page of trace. 

An associated issue that should be a ticket for the GUI is that clicking the "Rekey" button, while generating pages of log entries, does not give any user feedback of any sort. :(
